web/client/src/library/pages/ide/context.tsx

```diff
@@ -1,13 +1,46 @@
 import { createContext, type ReactNode, useState, useContext } from 'react'
 
+export const EnumErrorKey = {
+  General: 'general',
+  RunPlan: 'run-plan',
+  Environments: 'environments',
+  Models: 'models',
+  ApplyPlan: 'apply-plan',
+  Fetchdf: 'fetchdf',
+  EvaluateModel: 'evaluate-model',
+  RenderModel: 'render-model',
+  ColumnLineage: 'column-lineage',
+} as const
+
+export type ErrorKey = (typeof EnumErrorKey)[keyof typeof EnumErrorKey]
+
+export interface ErrorIDE {
```

Comment:
Is there a way to have this in the openapi.json spec so we won't forget to change this if we change the error dict in the server? Maybe convert the error into a Pydantic model and use it in the error handler response so FastAPI can pick it up. I'm not sure if it'd work that smoothly, but just an idea.

Reply:
I've added model `ApiExceptionPayload` and included it openapi generated output